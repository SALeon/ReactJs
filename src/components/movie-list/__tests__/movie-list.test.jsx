import React from 'react';
import { shallow } from 'enzyme';
import MovieList from '../index';

const mockProps = [
  {
    id: 'id-1',
    title: 'title-1',
    release_date: 'release_date-1',
    poster_path: 'poster_path-1',
    genres: ['genres-1'],
    vote_count: 1,
  },
  {
    id: 'id-2',
    title: 'title-2',
    release_date: 'release_date-2',
    poster_path: 'poster_path-2',
    genres: ['genres-2'],
    vote_count: 2,
  },
  {
    id: 'id-3',
    title: 'title-3',
    release_date: 'release_date-3',
    poster_path: 'poster_path-3',
    genres: ['genres-3'],
    vote_count: 3,
  },
];

it('MovieList snapshot', () => {
  const wrapper = shallow(<MovieList movies={mockProps} />);
  expect(wrapper).toMatchSnapshot();
});
