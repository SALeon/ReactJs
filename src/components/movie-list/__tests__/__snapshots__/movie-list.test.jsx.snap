// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList snapshot 1`] = `
<ul
  className="movie-list"
>
  <li
    className="movie-list__item"
    key="id-1"
  >
    <MovieCard
      movieInfo={
        Object {
          "genres": Array [
            "genres-1",
          ],
          "id": "id-1",
          "poster": "poster_path-1",
          "releaseDate": "release_date-1",
          "title": "title-1",
          "voteCount": 1,
        }
      }
    />
  </li>
  <li
    className="movie-list__item"
    key="id-2"
  >
    <MovieCard
      movieInfo={
        Object {
          "genres": Array [
            "genres-2",
          ],
          "id": "id-2",
          "poster": "poster_path-2",
          "releaseDate": "release_date-2",
          "title": "title-2",
          "voteCount": 2,
        }
      }
    />
  </li>
  <li
    className="movie-list__item"
    key="id-3"
  >
    <MovieCard
      movieInfo={
        Object {
          "genres": Array [
            "genres-3",
          ],
          "id": "id-3",
          "poster": "poster_path-3",
          "releaseDate": "release_date-3",
          "title": "title-3",
          "voteCount": 3,
        }
      }
    />
  </li>
</ul>
`;
